.github/workflows/pull-request.yml

```diff
@@ -185,7 +185,11 @@ jobs:
 
   verify:
     runs-on: ubuntu-latest
+    if: always()
     needs: ['check', 'integration-test', 'visual-test']
     steps:
-      - name: Done
-        run: echo "Done"
+      - name: Failure Test
+        if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') # Test the result of all dependencies
+        run: exit 1
+      - name: Success
+        run: exit 0
```

Comment:
The `always` forces the `verify` job to run no matter if dependent jobs have failed. This is required because Github Actions skip jobs when a failure happens. Also Github status checks consider a skip to be a success.

The `verify` job is a required check for all PRs, so it should always be run. The `if: contains ...` with the `exit 1` ensures a proper failure if any dependent job has a result of `failure` or `cancelled`.